Matrix J
Matrix xDot
---
Matrix qDot